specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2018-05-01/componentProactiveDetection_API.json

```diff
@@ -0,0 +1,315 @@
+        "operationId": "ProactiveDetectionConfigurations_Update",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ConfigurationId"
+          },
+          {
+            "name": "ProactiveDetectionProperties",
+            "description": "Properties that need to be specified to update the ProactiveDetection configuration.",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ApplicationInsightsComponentProactiveDetectionConfiguration"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The ProactiveDetection configuration that was successfully updated.",
+            "schema": {
+              "$ref": "#/definitions/ApplicationInsightsComponentProactiveDetectionConfiguration"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "ProactiveDetectionConfigurationUpdate": {
+            "$ref": "./examples/ProactiveDetectionConfigurationUpdate.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ApplicationInsightsComponentProactiveDetectionConfigurationListResult": {
+      "description": "A list of ProactiveDetection configurations.",
+      "type": "array",
+      "items": {
+        "$ref": "#/definitions/ApplicationInsightsComponentProactiveDetectionConfiguration"
+      }
+    },
+    "ApplicationInsightsComponentProactiveDetectionConfiguration": {
+      "description": "A ProactiveDetection configuration definition.",
+      "type": "object",
+      "x-ms-azure-resource": true,
+      "properties": {
+        "id": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Azure resource Id"
+        },
+        "name": {
+          "type": "string",
+          "description": "Azure resource name"
+        },
+        "type": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Azure resource type"
+        },
+        "location": {
+          "type": "string",
+          "description": "Resource location"
+        },
+        "properties": {
+          "description": "Properties that define a ProactiveDetection configuration.",
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/ApplicationInsightsComponentProactiveDetectionConfigurationProperties"
+        }
+      }
+    },
+    "ApplicationInsightsComponentProactiveDetectionConfigurationProperties": {
+      "description": "Properties that define a ProactiveDetection configuration.",
+      "type": "object",
+      "properties": {
+        "Name": {
```

Comment:
how is this name different from the one outside of properties? If different, please call this something else like ruleName